src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+export enum AriaLivePoliteness {
+  off = 'off',
+  polite = 'polite',
+  assertive = 'assertive',
+}
+
+/**
+ * Time in milliseconds before inserting the content into the container
+ */
+export const ARIA_LIVE_TICK: number = 100;
+
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private ariaLiveElement: HTMLElement;
+  private document: Document;
+  private previousTimeout: ReturnType<typeof setTimeout>;
+  private id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(
+    private _ngZone: NgZone,
+    @Inject(DOCUMENT) _document: any,
+    @Inject(PLATFORM_ID) private platformId: Object
+  ) {
+    this.document = _document;
+  }
+
+  private createLiveAriaContainer(): HTMLElement {
+    const ariaLiveElement = this.document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', AriaLivePoliteness.polite);
+
+    this.document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string | HTMLElement, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): void {
+    // Create aria-live container when needed.
+    if (this.createLiveContainer() === false) {
+      // when there is not ariaLiveElement we could not apply text to nothing so
+      // that's why we gonna break here.
+      return;
+    }
+
+    // handle DOM or string input
+    message = typeof message !== 'string' && isPlatformBrowser(this.platformId) ? message.textContent : message;
+
+    // when there is no message do NOTHING!
+    if (!message) {
+      return;
+    }
+
+    this.clear();
```

Comment:
I really don't see why we need to clean up the content beforehand. Assistive technologies will read the entire content change. On top of that, we have added the following line:
```
ariaLiveElement.setAttribute('aria-atomic', 'true');
```
So whatever we add to the aria-live element, assistive technologies treat changes as one whole thing.
>Assistive technologies will present the entire changed region as a whole, including the author-defined label if one exists.
https://www.digitala11y.com/aria-atomic-properties/